lib/phlex/rails/sgml/state.rb

```diff
@@ -5,16 +5,19 @@ def buffer
 		@user_context[:capture_context]&.output_buffer&.raw_buffer || super
 	end
 
-	def around_capture
-		original_capturing = @capturing
-		original_fragments = @fragments
+	def capture
+		if (capture_context = @user_context[:capture_context])
+			original_capturing = @capturing
+			original_fragments = @fragments
 
-		begin
-			@capturing = true
-			yield
-		ensure
-			@capturing = original_capturing
-			@fragments = original_fragments
+			capture_context.capture do
+				yield
+			ensure
+				@capturing = original_capturing
+				@fragments = original_fragments
+			end
```

Comment:
I'm not sure I understand what's going on with the `@capturing` and `@fragments` here. Are they expected to be mutated inside the `yield`? Why are we saving their state and restoring it, but not (seemingly) making any changes to them?

Reply:
Ops, I missed something. Let me fix this.